compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangPackageBuilder.java

```diff
@@ -620,8 +619,14 @@ void addFunctionType(DiagnosticPos pos, Set<Whitespace> ws, boolean paramsAvail,
 
         if (paramsAvail) {
             functionTypeNode.addWS(commaWsStack.pop());
-            this.varListStack.pop().forEach(v -> functionTypeNode.params.add(v));
+            functionTypeNode.params.addAll(this.varListStack.pop());
+
+        }
+
+        if (restParamAvail) {
+            functionTypeNode.restParam = (BLangSimpleVariable) this.restParamStack.pop();
         }
+        functionTypeNode.flagSet.add(Flag.PUBLIC);
```

Comment:
Why are we adding this flag?

Reply:
This was discussed in length the code review :)
TLDR:
FunctionTypeName can be always public unless it is overridden when used as a type definition. (`private function(int, int) foo = something`)

This was a suggestion by @hasithaa